test/gateway/eip20_cogateway/redeem.js

```diff
@@ -0,0 +1,201 @@
+        gateway,
+    );
+    
+    await utilityToken.transfer(accounts[7], stakerBalance,{from: owner});
+    
+    await utilityToken.approve(
+        eip20CoGateway.address,
+        new BN(1000),
+        {from: accounts[7]},
+    );
+    
+}
+
+contract('EIP20CoGateway.redeem() ', function (accounts) {
+    
+    let amount,
+        beneficiary = accounts[4],
+        gasPrice = new BN(10),
+        gasLimit = new BN(10),
+        nonce = new BN(1),
+        hashLockObj = Utils.generateHashLock(),
+        hashLock = hashLockObj.l;
+    
+    beforeEach(async function () {
+        
+        await _setup(accounts);
+        amount = new BN(100);
+        
+    });
+    
+    it('should fail when bounty amount is not equal as msg.value', async function () {
+
+        let bounty = new BN(10);
+        await Utils.expectRevert(eip20CoGateway.redeem(
+            amount,
+            beneficiary,
+            gasPrice,
+            gasLimit,
+            nonce,
+            hashLock,
+            { from: accounts[7], value: bounty },
+            ),
+            'msg.value must match the bounty amount',
+        );
+    });
+    
+    it('should fail when amount is 0', async function () {
+
+        amount = 0;
+        await Utils.expectRevert(eip20CoGateway.redeem(
+            amount,
+            beneficiary,
+            gasPrice,
+            gasLimit,
+            nonce,
+            hashLock,
+            { from: accounts[7], value: bountyAmount },
+            ),
+            'Redeem amount must not be zero',
+        );
+    });
+    
+    it('should fail when messagehash is not in undeclared state', async function () {
+        
+        let messageHash = "0x193fa194eef3c001da102ee129c23b1e13a723cb9335edefe9100e85132c77d8";
+        let mockEIP20CoGateway = await MockEIP20CoGateway.new(
+            valueToken,
+            utilityToken.address,
+            core.address,
+            bountyAmount,
+            organization,
+            gateway,
+        );
+        
+        await mockEIP20CoGateway.setOutboxStatus(messageHash);
+        
+        await Utils.expectRevert(mockEIP20CoGateway.redeem(
+            amount,
+            beneficiary,
+            gasPrice,
+            gasLimit,
+            nonce,
+            hashLock,
+            { from: accounts[7], value: bountyAmount },
+            ),
+            'Message status must be Undeclared',
+        );
+        
+    });
+    
+    it('should successfully redeem', async function () {
+    
+        let messageHash = "0x193fa194eef3c001da102ee129c23b1e13a723cb9335edefe9100e85132c77d8";
+        
+        let response = await eip20CoGateway.redeem(
+            amount,
+            beneficiary,
+            gasPrice,
+            gasLimit,
+            nonce,
+            hashLock,
+            { from: accounts[7], value: bountyAmount },
+        );
+
+        assert.strictEqual(
+            (await utilityToken.balanceOf(eip20CoGateway.address)).cmp(amount),
```

Comment:
For readability, we can use `.eq`
`let balance = await utilityToken.balanceOf(eip20CoGateway.address)`
and
`assert.strictEqual(balance.eq(amount), true, <message>);`